test/integration/ethereum/morphoLeverageStrategyExtension.spec.ts

```diff
@@ -0,0 +1,5238 @@
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith("Slippage tolerance must be <100%");
+        });
+      });
+
+      describe("when incentivized slippage tolerance is >100%", async () => {
+        beforeEach(async () => {
+          subjectIncentiveSettings.incentivizedSlippageTolerance = ether(1.1);
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith(
+            "Incentivized slippage tolerance must be <100%",
+          );
+        });
+      });
+
+      describe("when incentivize leverage ratio is less than max leverage ratio", async () => {
+        beforeEach(async () => {
+          subjectIncentiveSettings.incentivizedLeverageRatio = ether(2.29);
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith(
+            "Incentivized leverage ratio must be > max leverage ratio",
+          );
+        });
+      });
+
+      describe("when rebalance interval is shorter than TWAP cooldown period", async () => {
+        beforeEach(async () => {
+          subjectMethodologySettings.rebalanceInterval = ZERO;
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith(
+            "Rebalance interval must be greater than TWAP cooldown period",
+          );
+        });
+      });
+
+      describe("when TWAP cooldown period is shorter than incentivized TWAP cooldown period", async () => {
+        beforeEach(async () => {
+          subjectExecutionSettings.twapCooldownPeriod = ZERO;
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith(
+            "TWAP cooldown must be greater than incentivized TWAP cooldown",
+          );
+        });
+      });
+
+      describe("when an exchange has a twapMaxTradeSize of 0", async () => {
+        beforeEach(async () => {
+          subjectExchangeSettings.twapMaxTradeSize = ZERO;
+        });
+
+        it("should revert", async () => {
+          await expect(subject()).to.be.revertedWith("Max TWAP trade size must not be 0");
+        });
+      });
+    });
+
+    describe("#engage", async () => {
+      let destinationTokenQuantity: BigNumber;
+      let subjectCaller: Account;
+      let subjectExchangeName: string;
+
+      context(
+        "when rebalance notional is greater than max trade size and greater than max borrow",
+        async () => {
+          let issueQuantity: BigNumber;
+
+          const intializeContracts = async () => {
+            await initializeRootScopeContracts();
+
+            await wsteth.approve(debtIssuanceModule.address, ether(1000));
+            // await usdc.approve(debtIssuanceModule.address, ether(1000));
+
+            // Issue 1 SetToken
+            issueQuantity = ether(1);
+
+            await morphoLeverageModule.sync(setToken.address, { gasLimit: 10000000 });
```

Comment:
I'm curious why you have to call `sync` here? I thought there would be no deposit or borrow positions in Morpho before calling engage.